addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerJsonParser.java

```diff
@@ -66,12 +72,33 @@ public WappalyzerJsonParser() {
 
     WappalyzerData parse(String categories, List<String> technologies, boolean createIcons) {
         LOGGER.info("Starting to parse Tech Detecction technologies.");
+        Instant start = Instant.now();
         WappalyzerData wappalyzerData = new WappalyzerData();
         parseCategories(wappalyzerData, getStringResource(categories));
-        technologies.forEach(
-                path -> parseJson(wappalyzerData, getStringResource(path), createIcons));
+        // Process the files/paths in parallel
+        ExecutorService executor =
+                Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() - 1);
+        List<CompletableFuture<?>> futures =
+                technologies.stream()
+                        .map(
+                                path ->
+                                        CompletableFuture.supplyAsync(
+                                                () ->
+                                                        parseJson(
+                                                                wappalyzerData,
+                                                                getStringResource(path),
+                                                                createIcons),
+                                                executor))
+                        .collect(Collectors.toList());
+        // Get all the finished tasks
+        @SuppressWarnings("unused")
+        List<?> results =
+                futures.stream().map(CompletableFuture::join).collect(Collectors.toList());
```

Comment:
Why collect twice?

Reply:
That was based on examples I was looking at all of which had a start and then collect finished. But maybe it isn't needed. Will test further.

Reply:
Leaving "futures" as a stream is actually markedly slower for some reason.

Reply:
Now just one collect.

Reply:
Maybe it's the use of Constants, which uses more threads.

Reply:
It should not need to collect, just do the forEach directly.

Reply:
Bump.

Reply:
For the record, having them split performs better.